packages/http-specs/specs/type/property/additional-properties/main.tsp

```diff
@@ -463,17 +463,19 @@ interface SpreadRecordUnion
     > {}
 
 // ****************** spread record of discriminated unions **************************
-@discriminated(#{ envelope: "none" })
-union WidgetData {
-  kind0: WidgetData0,
-  kind1: WidgetData1,
-}
+// @discriminated(#{ envelope: "none" })
+// union WidgetData {
+//   kind0: WidgetData0,
+//   kind1: WidgetData1,
+// }
 
 model WidgetData0 {
+  kind: "kind0";
   fooProp: string;
 }
 
 model WidgetData1 {
+  kind: "kind1";
   start: utcDateTime;
   end?: utcDateTime;
 }
```

Comment:
I've put back the `kind` property, for the 3 cases of "union without discriminated".